portality/forms/application_processors.py

```diff
@@ -282,13 +282,14 @@ def finalise(self, account, save_target=True, email_alert=True):
 
         # if this application is being accepted, then do the conversion to a journal
         if self.target.application_status == constants.APPLICATION_STATUS_ACCEPTED:
-            j = applicationService.accept_application(self.target, current_user._get_current_object())
+            j = applicationService.accept_application(self.target, account)
             # record the url the journal is available at in the admin are and alert the user
-            jurl = url_for("doaj.toc", identifier=j.toc_id)
-            if self.source.current_journal is not None:  # todo: are alerts displayed?
-                self.add_alert('<a href="{url}" target="_blank">Existing journal updated</a>.'.format(url=jurl))
-            else:
-                self.add_alert('<a href="{url}" target="_blank">New journal created</a>.'.format(url=jurl))
+            if has_request_context():
```

Comment:
I'm very conflicted by this.  I see why you need it, but it also makes me think that the abstraction is wrong somewhere - our core processing code shouldn't require a request context to run.  Don't know what to do about it, probably just have to keep this as-is for the moment, but a note that we should think again.

Reply:
I agree with you there, I thought the same thing when I was running it and it failed due to the alerts. This is where a notification processor would help, so that context doesn't interfere with the logic laid out in the processor.

Reply:
I think that's the right answer - it's separating the notification code from the form processing code that's required.  I would also like to make it, if possible, that the form processors are "below" the BLL level, so that they are independent of call context; I think that would make a variety of things cleaner and easier, so maybe that's something we can work towards.  Then the notifications system can spin up a test context if it needs to generate urls, etc.

Reply:
So, what I mean is, lets not change this now, but know how to change it in the future